### DISCUSSION TITLE
V3 build

### DISCUSSION THREAD

**Discussion by chaloney** [doudar/SmartSpin2k#451] [General]
I am in the process of building a v3 and  I managed to break the entire switch off the pcb.   I need to know what pcb leads to solder together to replicate the switch in the TX position to use on a peloton without using the peloton tablet. I found a .3mf of the pcb on GitHub and found an online 3rd viewer. Is there away I can solder a jumper wire between 2 of the leads to replicate the TX position of the switch?
![69798BA0-1D5C-4D2A-A984-8D6528999BFD](https://user-images.githubusercontent.com/79726549/217313328-5431de00-608c-4c66-8683-ff7e13e2c1c9.jpeg)

**Comment by doudar** [comment]
Yep! All you need to do (if you only ever want to use TX mode) is to bridge the two pads with traces together. All the switch does is physically connect the TX line to the UART IC. You can probably bridge all three together. The open pin is just a n/c so it shouldn't hurt anything. 

![image](https://user-images.githubusercontent.com/17362216/217315477-1f567668-81f3-41a0-99bd-6b06c791da7f.png)

Below is the current layout of the (more robust) switch for R3. You can use it for reference but the exact routing may have changed since your board revision. 

![image](https://user-images.githubusercontent.com/17362216/217315850-b4c26f30-7570-4984-b28f-988c2cb2aab4.png)


Sorry you got one of those first gen boards with the old switch! Hopefully @eMadman gave you a good deal on it :)

**Comment by chaloney** [comment]
I’ll just solder all three leads to be sure. Thanks

**Comment by chaloney** [comment]
Soldering across all three seems to have worked. 
![F3784C53-422C-49E7-9BA4-72C3DCA4E70A](https://user-images.githubusercontent.com/79726549/217607711-c412385a-b084-4d1f-aba1-f3febccb46f4.jpeg)

power and cadence from the peloton was transmitted throughout the ride. The SS2K  didn’t seem to react to the big changes in power requested in the TrainerRoad workout—had to turn it by hand each time. It did adjust from there fairly well. This is with the latest firmware—flashed last night 23.2.1. Is this a setting problem?  The only thing I changed was max power and stepper direction to get the web shifter turning the right way.

**Reply by chaloney** [reply]
Turns out it was not settings, it was mechanical. The gear was a little loose on the stepper motor so the shaft would slip at times. Working much better with some CA glue. Also the stepper could move around a bit in the case and may have been enough that the gears slipped some of the time. Filled the gap with some adhesive backed rubber strip and now it seems to be working perfectly with essentially the default settings
